### PR TITLE
ref: Refactor smartSearchBar to pass in getFieldDefinition

### PR DESCRIPTION
**Pull request by ryan953**
The idea is that the existing results from `getFieldDefinition()` are specific to data in the events table. For replays we have a different set of fields with their own definitions and meanings. 

I want to be able to keep the list of replay fields separate from event fields. If they are not separate then someone could add a tag like `duration` and then they will see the definition as it relates to replays -> that's not helpful because in reality it's just a user tag.

This is going to be put to use in https://github.com/getsentry/sentry/pull/40683

### CODE REVIEW

File: static/app/components/smartSearchBar/index.tsx [hunk 1]
```diff
@@ -132,134 +132,144 @@ export type ActionBarItem = {
...
-   */
-  actionBarItems?: ActionBarItem[];
-  className?: string;
-
-  /**
-   * Custom Performance Metrics for query string unit parsing
-   */
-  customPerformanceMetrics?: CustomMeasurementCollection;
-  defaultQuery?: string;
+type DefaultProps = {
```
Comment (ryan953) [inline static/app/components/smartSearchBar/index.tsx:135]: i hoisted some fields up, and added `fieldDefinitionGetter`

'hide whitespace' in github is helpful here.

File: static/app/components/smartSearchBar/index.tsx [hunk 2]
```diff
@@ -1073,15 +1084,23 @@ class SmartSearchBar extends Component<Props, State> {
     if (searchTerm) {
       const preparedSearchTerm = prepareQuery ? prepareQuery(searchTerm) : searchTerm;
 
-      tagKeys = filterKeysFromQuery(tagKeys, preparedSearchTerm);
+      tagKeys = filterKeysFromQuery(
+        tagKeys,
+        preparedSearchTerm,
+        this.props.fieldDefinitionGetter
```
Comment (billyvg) [inline static/app/components/smartSearchBar/index.tsx:1090]: nit, can we destructure `fieldDefinitionGetter`

**Review by billyvg — APPROVED** [review]

File: static/app/components/smartSearchBar/index.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/components/smartSearchBar/index.tsx]: yeah, done!

File: static/app/components/smartSearchBar/index.tsx [hunk 3]
```diff
@@ -1063,31 +1074,38 @@ class SmartSearchBar extends Component<Props, State> {
    * Returns array of possible key values that substring match `query`
    */
   getTagKeys(searchTerm: string): [SearchItem[], ItemType] {
-    const {prepareQuery, supportedTagType} = this.props;
-
-    const supportedTags = this.props.supportedTags ?? {};
-    const {excludedTags} = this.props;
```
Comment (billyvg) [inline static/app/components/smartSearchBar/index.tsx:1069]: Thanks for cleaning this up too! 👯

**Review by billyvg — APPROVED** [review]

**Pull request merged**